src/status_im2/contexts/activity_center/notification/contact_requests/view.cljs

```diff
@@ -0,0 +1,113 @@
+        :icon      :main-icons2/add-user
+        :timestamp (datetime/timestamp->relative (:timestamp notification))
+        :unread?   (not (:read notification))
+        :context   [(i18n/label :t/contact-request-outgoing)
+                    [common/user-avatar-tag chat-id]]
+        :message   {:body (get-in message [:content :text])}
+        :items     (case contact-request-state
+                     constants/contact-request-state-mutual
+                     [{:type                :button
+                       :subtype             :danger
+                       :label               (i18n/label :t/cancel)
+                       :accessibility-label :cancel-contact-request
+                       :on-press            (fn []
+                                              (rf/dispatch
+                                               [:activity-center.contact-requests/cancel-outgoing-request
+                                                (:from message)])
+                                              (rf/dispatch [:activity-center.notifications/mark-as-read
+                                                            id]))}
+                      {:type    :status
+                       :subtype :pending
+                       :label   (i18n/label :t/pending)}]
+
+                     constants/contact-request-message-state-declined
+                     [{:type    :status
+                       :subtype :pending
+                       :label   (i18n/label :t/pending)}]
+
+                     nil)}])))
+
+(defn incoming-contact-request-view
+  [{:keys [id author message last-message] :as notification}]
+  (let [message (or message last-message)]
+    [quo/activity-log
+     {:title     (i18n/label :t/contact-request)
+      :icon      :main-icons2/add-user
+      :timestamp (datetime/timestamp->relative (:timestamp notification))
+      :unread?   (not (:read notification))
+      :context   [[common/user-avatar-tag author]
+                  (i18n/label :t/contact-request-sent)]
+      :message   {:body (get-in message [:content :text])}
+      :items
+      (case (:contact-request-state message)
+        constants/contact-request-message-state-accepted
+        [{:type    :status
+          :subtype :positive
+          :label   (i18n/label :t/accepted)}]
+
+        constants/contact-request-message-state-declined
+        [{:type    :status
+          :subtype :negative
+          :label   (i18n/label :t/declined)}]
+
+        constants/contact-request-state-mutual
+        [{:type                :button
+          :subtype             :danger
+          :label               (i18n/label :t/decline)
+          :accessibility-label :decline-contact-request
+          :on-press            (fn []
+                                 (rf/dispatch [:activity-center.contact-requests/decline-request id])
+                                 (rf/dispatch [:activity-center.notifications/mark-as-read
```

Comment:
FYI: Recently @MishkaRogachev opened a PR in status-go to automatically mark notifications with call to action as read. Soon we'll be able to update all places in status-mobile. We recently had an issue due to concurrent dispatches to do X + mark as read.